### ISSUE TITLE
Unable to test CSS User Origin in an automated way

### ISSUE THREAD

**Issue by zcorpan** [web-platform-tests/wpt#23980]
As far as I know, it's not possible to automate tests to exercise https://drafts.csswg.org/css-cascade-4/#cascade-origin-user

This affects the 'appearance' property, which has special handling with the User Agent Origin and User Origins. https://github.com/whatwg/html/pull/4857

cc @stephenmcgruer @frivoal

**Comment by stephenmcgruer** [comment]
So I'm clear: would a workable solution to this be a webdriver extension which could tell the browser what style to provide, probably for both the User Origin and User Agent Origin?

**Comment by zcorpan** [comment]
I believe that would work, yes. cc @AutomatedTester @shs96c

**Comment by zcorpan** [comment]
For User Agent Origin, the UA stylesheet already exists (and is specified in the HTML standard), and anything that could theoretically be affected by UA origin but isn't in the UA stylesheet seems to me it's in the realm of things that are not interesting to test, and could interfere with what is possible to optimize. So it might be better to *not* provide anything for User Agent Origin.

**Comment by frivoal** [comment]
Agreed, for the **UA** layer, it doesn't matter if it's coming from an actual stylesheet or if it has the same results through some other way, as long as the effects are the same, and that can be tested today, by checking if the expected effects are there. There's limited interest in inserting different UA stylesheet to see if it would be well behaved.

Do you think there's anything aout the `appearance` property that cannot be tested by merely observing its effects on the Author layer?

For the User origin, it is more interesting. I do think it'd be good to have some testing infrastructure to be able to provide a stylesheet, and have it side loaded at the User origin.

**Comment by zcorpan** [comment]
Yes, this: https://github.com/whatwg/html/pull/4857/files#diff-36cd38f49b9afa08222c0dc9ebfe35ebR112457

For example, with this user stylesheet:

```
input[type=text] { background: yellow; }
```

does it result in a "none" widget or "textfield" widget

**Comment by frivoal** [comment]
That's an example of needing the User origin, for which I agree. Is there an need to access the UA origin?

**Comment by zcorpan** [comment]
No
